core/domain/classifier_services.py

```diff
@@ -214,6 +214,13 @@ def save_classifier(classifier):
         classifier_id = _create_classifier(classifier)
     else:
         _update_classifier(classifier_data_model, classifier.state_name)
+
+    # Create the mapping from <exp_id,exp_version,state_name>
+    # to classifier_id.
+    create_classifier_exploration_mapping(
```

Comment:
The way you propose it does stray from the purpose of the method and, for that reason, I wouldn't do it like that. Maybe, in the re-training-is-false case, just call the mapping function directly.